Automattic/sensei
# Add course theme onboarding

Part of #4386

### Changes proposed in this Pull Request

* It introduces a new onboarding explaining the learning mode when the course editor is loaded.
* In order to introduce a new onboarding, some changes were needed also in the WCPC product tour: 719-gh-Automattic/sensei-wc-paid-courses because previously we were considering that we could have only the Gutenberg onboarding.
* I kept the same "X" we have in other onboardings to allow the user to close it without enabling the learning mode.
* It contains a temporary link to Learn More, going to [Sensei LMS documentation](https://senseilms.com/docs/), but it will be updated when we have the [specific documentation page for the Learning mode](https://github.com/Automattic/sensei/issues/4489).
* When clicking on "Enable learning mode", it updates the setting, open the course theme panel on the sidebar, and saves the post. I started thinking that saving the post would be a little invasive, but since it only happens immediately after the page is loaded, the user wouldn't have changes in the editor yet. If it's a new post, it saves a draft. Otherwise, the user would need to save the post manually.

### Testing instructions


* Enable the feature flag: `add_filter( 'sensei_feature_flag_course_theme', '__return_true' );`
* Activate WCPC plugin, and checkout this branch: 719-gh-Automattic/sensei-wc-paid-courses  to test it together.
* Clear the local storage for the WP site.
* Make sure you can navigate through all the 3 onboardings (Gutenberg, product tour, and learning mode).
* Also, make sure that when you click on "Enable learning mode", it opens the Learning mode in the sidebar, and enables it. Notice that the sidebar is being changed [here](https://github.com/Automattic/sensei/pull/4565) (it has a new title in the sidebar that is "Course Styles", which appears in the onboarding copy too).


### Screenshot / Video

A scenario where the user installed the WP and Sensei with WCPC, and is accessing the course editor for the first time.

https://user-images.githubusercontent.com/876340/148449209-50c5fd83-4ea4-42c6-8f9b-407c33957567.mov

A scenario where the user already saw the Gutenberg onboarding in the past, and they are accessing the course editor for the first time. In this case, the user decides to not enable the learning mode.

https://user-images.githubusercontent.com/876340/148449259-0468928d-49b3-4437-9b6c-b229590f2016.mov

A scenario where the user already uses Sensei, and they are accessing the course editor for the first time after Sensei update.

https://user-images.githubusercontent.com/876340/148449289-a14942ec-4616-4235-9624-51899b26c0e6.mov
